src/commands/trialApp/importTrialApp.ts

```diff
@@ -15,7 +16,22 @@ export async function importTrialApp(_context: IActionContext, loginSession: str
     await window.withProgress({ location: ProgressLocation.Notification, cancellable: false }, async p => {
         p.report({ message: localize('importingTrialApp', 'Importing trial app...') });
         ext.azureAccountTreeItem.trialAppNode = await TrialAppTreeItem.createTrialAppTreeItem(ext.azureAccountTreeItem, loginSession);
-        ext.context.globalState.update(TrialAppLoginSession, loginSession);
+        const trialAppNode = ext.azureAccountTreeItem.trialAppNode;
+
+        if (trialAppNode.client.metadata.timeLeft === undefined) {
+            throw new Error(localize('trialAppExpired', 'Trial app is expired. Could not import.'));
+        }
+
+        const expirationDate: Date = new Date();
```

Comment:
Use Date.now()  which returns the current time in ms.  Then just add the timeLeft * 1000